File: go/3038.go
package main

func maxOperations(nums []int) (ans int) {
	i := 0
	tmp := -1
	for i+1 < len(nums) {
		if tmp < 0 {
			tmp = nums[i] + nums[i+1]
			ans++
		} else if tmp == nums[i]+nums[i+1] {
			ans++
		} else {
			break
		}
		i += 2
	}
	return
}

File: python/3038.py
from typing import List

class Solution:
    def maxOperations(self, nums: List[int]) -> int:
        ans = 0
        i = 0
        tmp = -1
        while i + 1 < len(nums):
            if tmp < 0:
                tmp = nums[i] + nums[i+1]
                ans += 1
            elif tmp == nums[i] + nums[i+1]:
                ans += 1
            else:
                break
            i += 2
        return ans